### DISCUSSION TITLE
How to stream diarization and transcription (Whisper) results from an audio file?

### DISCUSSION THREAD

**Discussion by ColtonBehannon** [juanmc2005/diart#211] [Q&A] [answered]
Do you have any suggestions on how streaming results from an audio file can be supported in conjunction with transcription? I have read your article [here](https://betterprogramming.pub/color-your-captions-streamlining-live-transcriptions-with-diart-and-openais-whisper-6203350234ef) but this appears to be a bit outdated from where the library has progressed and also only demonstrates use from a mic. I am able to stream transcription results already and would like to be able to just add in speaker diarization.

**Comment by juanmc2005** [comment, marked as answer]
Hi @ColtonBehannon, if you already have streaming transcriptions (with timestamps), adding diarization would be a matter of running both in parallel and then aligning the output according to both sets of timestamps. I recently implemented a now outdated SpeakerAwareTranscription pipeline (see #147), but it's a bit hacky and should be improved before integrating that into the library.

Concerning the blog post, the same principles apply to the latest version of diart, only some names need to be updated. Otherwise you can simply install a previous version like v0.6 or v0.7.

**Reply by ColtonBehannon** [reply]
Thanks for the response, after leaving the comment I was actually able to get a basic implementation going by just changing some of the names from the blog post. I will check out #147 as well. Thanks!
